### DISCUSSION TITLE
Error: The Root Directory setting is not supported for vercel-sapper

### DISCUSSION THREAD

**Discussion by mikenikles** [vercel/vercel#5633] [Help]
Hi,

I have been using [`vercel-sapper`](https://github.com/thgh/vercel-sapper) for a year or more, but some time between early December 2020 and now something changed and deployments now fail with the following error in the build logs: `Error: The Root Directory setting is not supported for vercel-sapper`.

The full build logs:

```
19:51:56.378 | Cloning completed in 1179ms
19:51:56.381 | Analyzing source code...
19:51:56.383 | Warning: Due to `builds` existing in your configuration file, the Build and Development Settings defined in your Project Settings will not apply. Learn More: https://vercel.link/unused-build-settings
19:51:56.717 | Error: The Root Directory setting is not supported for vercel-sapper
```

My web app is located in a monorepo at `services/website` and the `services/website/vercel.json` file looks as follows ([source code](https://github.com/mikenikles/your-analytics/blob/master/services/website/vercel.json)):

```
{
  "version": 2,
  "builds": [
    {
      "src": "package.json",
      "use": "vercel-sapper"
    }
  ],
  "routes": [
    {
      "src": "/api/admin/(.*)",
      "dest": "https://admin-api.your-analytics.org/$1"
    },
    {
      "src": "/api/query/(.*)",
      "dest": "https://query-api.your-analytics.org/$1"
    }
  ]
}
```

For reference, there is also an open issue at https://github.com/thgh/vercel-sapper/issues/60.

Does anyone have insight into what changed in December?

Thanks,

Mike

**Comment by styfle** [comment]
Did you add "Root Directory" in your project settings on the Vercel Dashboard in December?

**Reply by mikenikles** [reply]
No, the "Root Directory" was configured six+ months earlier. It's been a monorepo from the beginning. The last deployment I did with code changes was in September, then there were a few automated deployments due to Dependabot. The last such deployment was in December and a few days ago, I wanted to push new code and got the error.

Quite odd... `vercel-sapper` does use some deprecated fields in `vercel.json`, maybe something changed that requires a change in `vercel.json` to support the root directory 🤔. Given it used to work, I'm not sure where to start looking though.

**Reply by antony** [reply]
Hey @mikenikles I believe it's because we have to override a fair few bits of the build configuration for Sapper. I've commented in more depth on the linked issue!

**Reply by mikenikles** [reply]
Yeah I'm on the same page, this is no longer an issue. Migration to SvelteKit is straightforward enough to make this the recommended best practice.
